eduNEXT/eox-nelp
# And/stats api view

## Description

This add a new endpoint in `/eox-nelp/api/stats/v1/tenant-stats/` that will return the basic tenant stats in a json format

```
{
    "learners": 1,
    "courses": {
        "total_courses": 4,
        "metrics": [
            {
                "id": "course-v1:patata+CS102+2023",
                "name": "PROCEDURAL SEDATION AND ANALGESIA COURSE",
                "learners": 0,
                "instructors": 1,
                "sections": 18,
                "sub_sections": 144,
                "units": 184,
                "components": {
                    "discussion": 0,
                    "drag-and-drop-v2": 0,
                    "html": 133,
                    "openassessment": 0,
                    "problem": 49,
                    "video": 0
                }
            },
            {
                "id": "course-v1:test+CS101+2022_T3",
                "name": "test",
                "learners": 1,
                "instructors": 0,
                "sections": 1,
                "sub_sections": 6,
                "units": 7,
                "components": {
                    "discussion": 0,
                    "drag-and-drop-v2": 2,
                    "html": 0,
                    "openassessment": 0,
                    "problem": 6,
                    "video": 2
                }
            },
            ...
        ]
    },
    "instructors": 1
}
```

issue https://edunext.atlassian.net/browse/FUTUREX-440

## Testing instructions
1. Go to the endpoint `/eox-nelp/api/stats/v1/tenant-stats/` and check 
2. To add components set the setting ALLOWED_VERTICAL_BLOCK_TYPES
